node/pkg/utils/request.go

```diff
@@ -78,6 +78,42 @@ func UrlRequest[T any](urlEndpoint string, method string, requestBody interface{
 		return result, errors.New("failed to make request")
 	}
 
+	resultBody, err := io.ReadAll(response.Body)
+	if err != nil {
+		log.Error().Err(err).Msg("failed to read response body")
+
+		return result, err
+	}
+
+	err = json.Unmarshal(resultBody, &result)
+	if err != nil {
+		log.Error().Err(err).Str("resultBody", string(resultBody)).Msg("failed to unmarshal response body")
+		return result, err
+	}
+
+	return result, nil
+}
+
+func GetRequestProxy[T any](urlEndpoint string, requestBody interface{}, headers map[string]string, proxy string) (T, error) {
+	return UrlRequestProxy[T](urlEndpoint, "GET", requestBody, headers, proxy)
+}
+
+func UrlRequestProxy[T any](urlEndpoint string, method string, requestBody interface{}, headers map[string]string, proxy string) (T, error) {
```

Comment:
Can we consolidate this function into on `urlRequest`, flagged by a parameter? Or is there any reason why we have to split it?

Reply:
I personally prefer having separate functions rather than increasing parameters, but I think I can reduce duplications, thank you for the feedback. I'll make an update